src/api/interfaces/state.rs

```diff
@@ -0,0 +1,291 @@
+/*
+   Copyright 2019 Supercomputing Systems AG
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+	   http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+use crate::{api::ApiResult, rpc::Subscribe, utils, Api, MetadataError, ReadProof, Request};
+use ac_compose_macros::rpc_params;
+pub use ac_node_api::{events::EventDetails, StaticEvent};
+use ac_primitives::{ExtrinsicParams, FrameSystemConfig};
+use codec::{Decode, Encode};
+use log::*;
+use serde::de::DeserializeOwned;
+use sp_core::storage::{StorageChangeSet, StorageData, StorageKey};
+
+/// Generic interface to substrate storage.
+pub trait GetState<Hash> {
```

Comment:
I would change this to `GetStorage`

Reply:
There's already a GetStorage in node_api to retrieve metadata storage values. I'd rather avoid having two equally named storage.

Two options, if you don't like `GetState`:
- rename the `GetStorage` from node api to `GetStorageTypes`
- rename `GetState` to `GetStateStorage` 

or similar naming.. what do you think?

Reply:
Went for the first one now - I think `GetStorageTypes` makes more sense anyhow.

Reply:
Yes, I agree! 👍